core/src/main/java/org/eclipse/hono/adapter/AdapterConfig.java

```diff
@@ -27,6 +30,10 @@
 
     private final Vertx vertx = Vertx.vertx();
 
+    @Autowired(required=false)
+    @Qualifier("registration")
+    private ClientConfigProperties registrationServiceClientConfig;
```

Comment:
That is needed for handling the optional registration service ClientConfigProperties/ConnectionFactory/HonoClient beans.

If no "hono.registration.host" property is set, the `registrationServiceClientConfig` bean will not be there.

For this case, different implementations of the `registrationServiceConnectionFactory()` method would cause problems:
>     @Qualifier("registration")
>     @Bean
>     public ConnectionFactory registrationServiceConnectionFactory() {
>         return new ConnectionFactoryImpl(registrationServiceClientConfig());
>     }
-> Exception on start (bean named 'registrationServiceClientConfig' could not be found)


>     @Qualifier("registration")
>     @Bean
>     public ConnectionFactory registrationServiceConnectionFactory(final ClientConfigProperties registrationServiceClientConfig) {
>         return new ConnectionFactoryImpl(registrationServiceClientConfig);
>     }
-> uses `ClientConfigProperties` object for HonoClient not RegistrationServiceClient


>     @Qualifier("registration")
>     @Bean
>     public ConnectionFactory registrationServiceConnectionFactory(@Qualifier("registration") final ClientConfigProperties registrationServiceClientConfig) {
>         return new ConnectionFactoryImpl(registrationServiceClientConfig);
>     }
-> Exception on start (bean of type 'org.eclipse.hono.config.ClientConfigProperties' could not be found)

So it's basically about `@Autowired(required=false)` not being applicable to method parameters.